### ISSUE TITLE
The retrained results have color differences and unremovable mask shadows.

### ISSUE THREAD

**Issue by xuanandsix** [yisol/IDM-VTON#145]
![WeChat73ba3211726c21a91433398b4f3885bb](https://github.com/user-attachments/assets/2153c7a4-9e78-4c36-87e2-251e0f61d877)

some suggestions, please.

**Comment by xuanandsix** [comment]
> How many cards did you use for training? Will NCCL timeout occur?

6 * A800， NCCL is work.  More training steps improve the color and mask problem.

**Comment by all1new** [comment]
What is the final effect? ​​Will it be better than the original weight of IDM-VTON?

**Comment by awais-nayyar** [comment]
I want to train this code on my own dataset. What is the minimum GPU and Ram requirements??? 
can anybody help to guide me?

**Comment by xuanandsix** [comment]
> What is the final effect? ​​Will it be better than the original weight of IDM-VTON?

For some pictures, there is still color difference. The color of the clothes is not as bright as the image prompt.

![WeChat7147d970e0627b928e0b22fad41261c8](https://github.com/user-attachments/assets/73d409d6-a519-4380-8517-157cb40fa5fb)

**Comment by all1new** [comment]
> > What is the final effect? ​​Will it be better than the original weight of IDM-VTON?
> 
> For some pictures, there is still color difference. The color of the clothes is not as bright as the image prompt.
> 
> ![WeChat7147d970e0627b928e0b22fad41261c8](https://private-user-images.githubusercontent.com/70092788/364630542-73d409d6-a519-4380-8517-157cb40fa5fb.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1MTMxNTcsIm5iZiI6MTcyNTUxMjg1NywicGF0aCI6Ii83MDA5Mjc4OC8zNjQ2MzA1NDItNzNkNDA5ZDYtYTUxOS00MzgwLTg1MTctMTU3Y2I0MGZhNWZiLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA1VDA1MDczN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3MDNlMWQ4OWE3YmVjOTM5YTM1NTBkNDY5NTRhNjRlN2Q3YjA1MTA1NzNhMTNhY2MzMmE1ZWNjNmI4ZjkwNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ugOTclhxTxEA9jtt3JyLtoMlC9T_9sUl8ad-fZ-5AJ4)

The conclusion of my training is the same as yours. Is there any way to improve it?

**Comment by all1new** [comment]
![image](https://github.com/user-attachments/assets/c7d0305e-1193-4564-aa06-dcd3f7ce8fd8)
Have you ever used the training code to do fine tuning? The initial inference result of the original weights is very strange. I wonder if there is a problem with the inference process in the training code?

**Comment by TnoobT** [comment]
> > > What is the final effect? ​​Will it be better than the original weight of IDM-VTON?
> > 
> > 
> > For some pictures, there is still color difference. The color of the clothes is not as bright as the image prompt.
> > ![WeChat7147d970e0627b928e0b22fad41261c8](https://private-user-images.githubusercontent.com/70092788/364630542-73d409d6-a519-4380-8517-157cb40fa5fb.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1MTMxNTcsIm5iZiI6MTcyNTUxMjg1NywicGF0aCI6Ii83MDA5Mjc4OC8zNjQ2MzA1NDItNzNkNDA5ZDYtYTUxOS00MzgwLTg1MTctMTU3Y2I0MGZhNWZiLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA1VDA1MDczN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3MDNlMWQ4OWE3YmVjOTM5YTM1NTBkNDY5NTRhNjRlN2Q3YjA1MTA1NzNhMTNhY2MzMmE1ZWNjNmI4ZjkwNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ugOTclhxTxEA9jtt3JyLtoMlC9T_9sUl8ad-fZ-5AJ4)
> 
> The conclusion of my training is the same as yours. Is there any way to improve it?

问题解决了吗，我训练了一天还是这样，loss感觉也没收敛

**Comment by TnoobT** [comment]
> I have encountered the same problem. Has anyone solved this problem? Can you give me some advice? I would be very grateful.

1. 不要用8bit adam优化器；
2. 用他的训练代码；https://github.com/luxiaolili/IDM-VTON-train

**Comment by jiayoujiayoujiayoua** [comment]
> > I have encountered the same problem. Has anyone solved this problem? Can you give me some advice? I would be very grateful.
> 
> 1. 不要用8bit adam优化器；
> 2. 用他的训练代码；https://github.com/luxiaolili/IDM-VTON-train
非常感谢你，我马上再训练一下试试，好人一生平安！

**Comment by all1new** [comment]
> > > I have encountered the same problem. Has anyone solved this problem? Can you give me some advice? I would be very grateful.
> > 
> > 
> > 
> > 1. 不要用8bit adam优化器；
> > 2. 用他的训练代码；https://github.com/luxiaolili/IDM-VTON-train
> >    非常感谢你，我马上再训练一下试试，好人一生平安！

我用这个尝试过, 速度有些慢, 从基础模型开始训练试一下, 我这边还是没法复现模型效果

**Comment by jiayoujiayoujiayoua** [comment]
好的佬，我主要是在自己的数据集上面做微调 不知道效果咋样 佬 加个联系方式吗 我训练有结果了一起讨论讨论





------------------ Original ------------------
From: 1_corn ***@***.***&gt;
Date: Wed,Sep 25,2024 4:54 PM
To: yisol/IDM-VTON ***@***.***&gt;
Cc: jiayoujiayoujiayoua ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [yisol/IDM-VTON] The retrained results have color differences andunremovable mask shadows. (Issue #145)





    
I have encountered the same problem. Has anyone solved this problem? Can you give me some advice? I would be very grateful.
   
不要用8bit adam优化器；
 
用他的训练代码；https://github.com/luxiaolili/IDM-VTON-train
 非常感谢你，我马上再训练一下试试，好人一生平安！
    
我用这个尝试过, 速度有些慢, 从基础模型开始训练试一下, 我这边还是没法复现模型效果

**Comment by TnoobT** [comment]
> > > > I have encountered the same problem. Has anyone solved this problem? Can you give me some advice? I would be very grateful.
> > > 
> > > 
> > > 
> > > 1. 不要用8bit adam优化器；
> > > 2. 用他的训练代码；https://github.com/luxiaolili/IDM-VTON-train
> > >    非常感谢你，我马上再训练一下试试，好人一生平安！
> 
> 我用这个尝试过, 速度有些慢, 从基础模型开始训练试一下, 我这边还是没法复现模型效果

还好吧，训5000个step就能inpainting了，整张图的色差也基本没有了

**Comment by all1new** [comment]
> > > > > I have encountered the same problem. Has anyone solved this problem? Can you give me some advice? I would be very grateful.
> > > > 
> > > > 
> > > > 
> > > > 1. 不要用8bit adam优化器；
> > > > 2. 用他的训练代码；https://github.com/luxiaolili/IDM-VTON-train
> > > >    非常感谢你，我马上再训练一下试试，好人一生平安！
> > 
> > 
> > 我用这个尝试过, 速度有些慢, 从基础模型开始训练试一下, 我这边还是没法复现模型效果
> 
> 还好吧，训5000个step就能inpainting了，整张图的色差也基本没有了

有没有对比过和IDM-VTON原始权重的效果, 以及对各种款式比如"毛衣", "图案复杂服饰", 线上模特; 这样多维度对比效果;
如果有的话 效果是否一致呢?

**Comment by all1new** [comment]
> 好的佬，我主要是在自己的数据集上面做微调 不知道效果咋样 佬 加个联系方式吗 我训练有结果了一起讨论讨论
> […](#)
> ------------------ Original ------------------ From: 1_corn ***@***.***&gt; Date: Wed,Sep 25,2024 4:54 PM To: yisol/IDM-VTON ***@***.***&gt; Cc: jiayoujiayoujiayoua ***@***.***&gt;, Comment ***@***.***&gt; Subject: Re: [yisol/IDM-VTON] The retrained results have color differences andunremovable mask shadows. (Issue #145) I have encountered the same problem. Has anyone solved this problem? Can you give me some advice? I would be very grateful. 不要用8bit adam优化器； 用他的训练代码；https://github.com/luxiaolili/IDM-VTON-train 非常感谢你，我马上再训练一下试试，好人一生平安！ 我用这个尝试过, 速度有些慢, 从基础模型开始训练试一下, 我这边还是没法复现模型效果 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

嗯嗯 加 我邮箱是1090207147@qq.com

**Comment by dittohed** [comment]
@xuanandsix @all1new @jiayoujiayoujiayoua 
Could you please share your experience with the training? Do you actually need to change anything compared to the training command from the README (apart from more training steps)? Is there a way to figure out early if the training is going right?

**Comment by mo230761** [comment]
@TnoobT @all1new @jiayoujiayoujiayoua 
佬，你们解决了这个问题嘛？要是我只是不使用8bita dam优化器有帮助吗

**Comment by ruanruano** [comment]
> [@TnoobT](https://github.com/TnoobT) [@all1new](https://github.com/all1new) [@jiayoujiayoujiayoua](https://github.com/jiayoujiayoujiayoua) 老兄，你们解决这个问题了吗？要是我只是不使用8bita dam优化器有帮助吗？

Sorry to bother you, hello, can you solve this problem by just not using 8bit? Is the problem solved?

**Comment by xuanandsix** [comment]
Based on my test results, when using the dataset from the original paper, extending the training time to approximately three times that of the original epoch can effectively resolve this issue.

**Comment by ruanruano** [comment]
> Based on my test results, when using the dataset from the original paper, extending the training time to approximately three times that of the original epoch can effectively resolve this issue.

Sorry to bother you again, should I train three times more epochs when using 8bit, or three times more epochs when not using 8bit?

**Comment by xuanandsix** [comment]
> > Based on my test results, when using the dataset from the original paper, extending the training time to approximately three times that of the original epoch can effectively resolve this issue.
> 
> Sorry to bother you again, should I train three times more epochs when using 8bit, or three times more epochs when not using 8bit?
Using 8bit，all settings remain unchanged.

**Comment by NH1900** [comment]
Hey @xuanandsix 

I have several questions appreciated your response with your most convenient time in advance:

1 For mixed_precision did you use fp16 or same as original train_xl.sh which did not assign any value to mixed_precision ending up in train_xl.py with float32?

2 Have you tried lower learning rate or dynamic learing rate?

3 For the final result with three times of epoch how close it is comparing to IDM official model? just double check?

**Comment by xuanandsix** [comment]
> Hey [@xuanandsix](https://github.com/xuanandsix)
> 
> I have several questions appreciated your response with your most convenient time in advance:
> 
> 1 For mixed_precision did you use fp16 or same as original train_xl.sh which did not assign any value to mixed_precision ending up in train_xl.py with float32?
> 
> 2 Have you tried lower learning rate or dynamic learing rate?
> 
> 3 For the final result with three times of epoch how close it is comparing to IDM official model? just double check?

1、same as original train_xl.sh.
2、I used the same learning rate when I used the original dataset, but I increased the learning rate when I expanded the dataset.
3、I think there are still some unknown tricks. I achieved better results by expanding the dataset.

**Comment by NH1900** [comment]
> > Hey [@xuanandsix](https://github.com/xuanandsix)
> > I have several questions appreciated your response with your most convenient time in advance:
> > 1 For mixed_precision did you use fp16 or same as original train_xl.sh which did not assign any value to mixed_precision ending up in train_xl.py with float32?
> > 2 Have you tried lower learning rate or dynamic learing rate?
> > 3 For the final result with three times of epoch how close it is comparing to IDM official model? just double check?
> 
> 1、same as original train_xl.sh. 2、I used the same learning rate when I used the original dataset, but I increased the learning rate when I expanded the dataset. 3、I think there are still some unknown tricks. I achieved better results by expanding the dataset.

Really thanks for replying to my previous questions. I have some followed up questions if you do not mind:
1 same with the current repo but not this repo right: https://github.com/luxiaolili/IDM-VTON-train/tree/main? 

the reason I am asking is for original repo the shell script did not have mixed_precision which means in python code it will be float 32 then memory and speed will be a problem; most importantly the dtype of sample input and Timesteps Module in diffusers lib in garment UNET are mismatch and error will occur. Not sure whether you met this.

2 But still fixed learning rate not dynamica learning rate? and when you mentioned expanded, does this mean you collect more training data pair?

3 True.

**Comment by xuanandsix** [comment]
> > > Hey [@xuanandsix](https://github.com/xuanandsix)
> > > I have several questions appreciated your response with your most convenient time in advance:
> > > 1 For mixed_precision did you use fp16 or same as original train_xl.sh which did not assign any value to mixed_precision ending up in train_xl.py with float32?
> > > 2 Have you tried lower learning rate or dynamic learing rate?
> > > 3 For the final result with three times of epoch how close it is comparing to IDM official model? just double check?
> > 
> > 
> > 1、same as original train_xl.sh. 2、I used the same learning rate when I used the original dataset, but I increased the learning rate when I expanded the dataset. 3、I think there are still some unknown tricks. I achieved better results by expanding the dataset.
> 
> Really thanks for replying to my previous questions. I have some followed up questions if you do not mind: 1 same with the current repo but not this repo right: https://github.com/luxiaolili/IDM-VTON-train/tree/main?
> 
> the reason I am asking is for original repo the shell script did not have mixed_precision which means in python code it will be float 32 then memory and speed will be a problem; most importantly the dtype of sample input and Timesteps Module in diffusers lib in garment UNET are mismatch and error will occur. Not sure whether you met this.
> 
> 2 But still fixed learning rate not dynamica learning rate? and when you mentioned expanded, does this mean you collect more training data pair?
> 
> 3 True.

Since I'm no longer engaged in research in this field at present, I have grown unfamiliar with the source code. What I can confirm is that I only adjusted those several hyperparameters that I mentioned previously.

**Comment by NH1900** [comment]
> > > > Hey [@xuanandsix](https://github.com/xuanandsix)
> > > > I have several questions appreciated your response with your most convenient time in advance:
> > > > 1 For mixed_precision did you use fp16 or same as original train_xl.sh which did not assign any value to mixed_precision ending up in train_xl.py with float32?
> > > > 2 Have you tried lower learning rate or dynamic learing rate?
> > > > 3 For the final result with three times of epoch how close it is comparing to IDM official model? just double check?
> > > 
> > > 
> > > 1、same as original train_xl.sh. 2、I used the same learning rate when I used the original dataset, but I increased the learning rate when I expanded the dataset. 3、I think there are still some unknown tricks. I achieved better results by expanding the dataset.
> > 
> > 
> > Really thanks for replying to my previous questions. I have some followed up questions if you do not mind: 1 same with the current repo but not this repo right: https://github.com/luxiaolili/IDM-VTON-train/tree/main?
> > the reason I am asking is for original repo the shell script did not have mixed_precision which means in python code it will be float 32 then memory and speed will be a problem; most importantly the dtype of sample input and Timesteps Module in diffusers lib in garment UNET are mismatch and error will occur. Not sure whether you met this.
> > 2 But still fixed learning rate not dynamica learning rate? and when you mentioned expanded, does this mean you collect more training data pair?
> > 3 True.
> 
> Since I'm no longer engaged in research in this field at present, I have grown unfamiliar with the source code. What I can confirm is that I only adjusted those several hyperparameters that I mentioned previously.

NP appreciated those information for real. Hope you have a wonderful day no matter where you are. Thanks!
